### ISSUE TITLE
Elevator problem on nether roof

### ISSUE THREAD

**Issue by nhanledev** [Zrips/CMI#8798]
### Description of Issue

After updating to 9.7.2.12, the Elevator sign does not allow players to go upward on any location above nether roof. It still works at a position in overworld and below nether roof.

### Version Information

```shell
[11:08:29 INFO]: --------------------------------------------------
[11:08:29 INFO]: CMI: 9.7.2.12 BungeeCord CMIB  MySQL
[11:08:29 INFO]: CMILib: 1.4.7.15 
[11:08:29 INFO]: Server: Paper(496) 1.20.4-R0.1-SNAPSHOT-
[11:08:29 INFO]: CMI economy: Off Vault: 1.7.3-CMI CMI Chat: Off
[11:08:29 INFO]: Modules -> 15 enabled 44 disabled: spawnerProximity, cuffed, customMessages, mirror, tablist, versionCheck, armorEffects, moneyCheque, skin, noTarget, elytraBoost, cmiPlaytime, elytraLaunch, bossBarCompass, coloredArmor, chatBubble, playerCombat, votifier, headDrop, attachedCommands, spawnerCharge, interactiveCommands, namePlates, timedCommands, homeInteractions, hpBossBar, launchPad, armorstand, portals, anvilRenameColor, jail, spawners, paintingEditor, dynamicSigns, playerChatTag, durabilityLoss, flightCharge, disabledEnchants, ranks, firstJoinMessages, selection, silentChest, warmup, totemBehavior
[11:08:29 INFO]: --------------------------------------------------
```

### Agreements

- [X] My server is supported by CMI.
- [X] My version of CMI at the time of this report is up to date.
- [X] I have searched the github and asked around before making this report.

**Comment by CloudeLecaw** [comment]
have you checked you havent enabled netherRoof protection on you server or adjust nether roof high?

**Comment by nhanledev** [comment]
> have you checked you havent enabled netherRoof protection on you server or adjust nether roof high?

No, the Elevator works in older version. I double checked the settings that you've mentioned and it looks good to me

```
  # Will teleport players down from nether 'roof'
  PreventPlayersOnNetherRoof: false
  # Will define nether roof height. Keeping it at 0 will use default nether height, which is 128
  netherRoofHeight: 0
  # This will teleport players up from beneath bedrock, if they ave flying or gliding with the elytra on.
  PreventPlayersBelowBedrock: true
  ```

**Comment by Zrips** [comment]
Retested elevators usage, and everything looks to be working correctly with regular players. By any chance there is another plugin interfering and preventing players teleportations?

**Comment by nhanledev** [comment]
> Retested elevators usage, and everything looks to be working correctly with regular players. By any chance there is another plugin interfering and preventing players teleportations?

I tested the elevator in multiple scenerios:
- In the overworld, survival mode: Work, Bothways
- In the nether, survival mode, below nether roof: Work, Bothways
- In the nether, survival mode, above nether roof: Work partially, downward only
- In the nether, CREATIVE mode, above nether roof: Work, Bothways

**Comment by nhanledev** [comment]
> > Retested elevators usage, and everything looks to be working correctly with regular players. By any chance there is another plugin interfering and preventing players teleportations?
> 
> I tested the elevator in multiple scenerios:
> 
> * In the overworld, survival mode: Work, Bothways
> * In the nether, survival mode, below nether roof: Work, Bothways
> * In the nether, survival mode, above nether roof: Work partially, downward only
> * In the nether, CREATIVE mode, above nether roof: Work, Bothways


I attach a video record that shows:

- Paper version
```
This server is running Paper version git-Paper-496 (MC: 1.20.4) (Implementing API version 1.20.4-R0.1-SNAPSHOT) (Git: 7ac24a1 on ver/1.20.4)
You are 1 version(s) behind
Download the new version at: https://papermc.io/downloads/paper
```

- CMI version
```
[01:40:15 INFO]: CMI: 9.7.2.12 SqLite
[01:40:15 INFO]: CMILib: 1.4.7.15
[01:40:15 INFO]: Server: Paper(496) 1.20.4-R0.1-SNAPSHOT-
[01:40:15 INFO]: CMI economy: Disabled CMI Chat: Disabled
[01:40:15 INFO]: Modules -> 58 enabled 1 disabled: votifier
[01:40:15 INFO]: --------------------------------------------------
```

- Current running plugins
```
[01:40:38 INFO]: Server Plugins (2):
[01:40:38 INFO]: Bukkit Plugins:
[01:40:38 INFO]:  - CMI, CMILib
```

The video: 


https://github.com/Zrips/CMI/assets/28459250/f1997590-83d4-4cf6-aa88-1b57c3e5caf1

**Comment by CloudeLecaw** [comment]
can you try to make the Plattform in the air two blocks deep instead of one row and recheck if the issue still remains?

**Comment by Zrips** [comment]
@nhanledev 
Could you test it without any other plugin to rule out 3rd party interference?

**Comment by nhanledev** [comment]
Hello, the test above is a fresh server with only CMI and CMILib. I will test it again with more than two rows of platform.

**Comment by nhanledev** [comment]
I attach a video record that has a 2-layer platform instead of one the eliminate the teleportation changes that CloudeLecaw has mentioned before. The issue is still persist.

The test server has only 2 plugins: CMI and CMILib, as shown in the beginning of the clip.

https://github.com/Zrips/CMI/assets/28459250/ef94bdfc-d340-4ecc-a013-28e3a1942467

**Comment by CloudeLecaw** [comment]
> @nhanledev Could you test it without any other plugin to rule out 3rd party interference?

he showed in his video he is using no other plugins. just cmilib and cmi.

**Comment by CloudeLecaw** [comment]
would you mind showing use your netherRoof setting?

**Comment by nhanledev** [comment]
> would you mind showing use your netherRoof setting?

Hello, that was the fresh server with default CMI settings, which has these in the config.yml file:

```
  # Will teleport players down from nether 'roof'
  PreventPlayersOnNetherRoof: false
  # Will define nether roof height. Keeping it at 0 will use default nether height, which is 128
  netherRoofHeight: 0
  # This will teleport players up from beneath bedrock, if they ave flying or gliding with the elytra on.
  PreventPlayersBelowBedrock: false
  ```

**Comment by Zrips** [comment]
@nhanledev 
Ok, thanks for info, reproduced issue now. Will make a fix with next version.
